info-compiler/src/test/java/com/google/sps/NewsContentExtractorTest.java

```diff
@@ -0,0 +1,200 @@
+    Metadata metadata = new Metadata();
+    Optional<NewsArticle> potentialNewsArticle =
+        newsContentExtractor.extractContentFromHtml(boilerpipeHandler, metadata, webpageStream,
+                                                    URL);
+    Assert.assertTrue(potentialNewsArticle.isPresent());
+    Assert.assertEquals(NEWS_ARTICLE_WITHOUT_TITLE, potentialNewsArticle.get());
+  }
+
+  @Test
+  public void extractContentFromHtml_emptyDocParser() throws IOException, SAXException,
+      TikaException {
+    // Extract content and meta data using a parser that modifies the content handler to
+    // receive an end-of-document notification via {@code endDocument()}. Therefore, an empty
+    // article will be extracted, while the URL is set correctly to the passed in parameter.
+    // Content processing hasn't occurred so the abbreivated content is null.
+    HtmlParser parser = mock(HtmlParser.class);
+    doAnswer(parseInvocation -> {
+               BoilerpipeContentHandler boilerpipeHandler =
+                   (BoilerpipeContentHandler) parseInvocation.getArguments()[1];
+               boilerpipeHandler.endDocument();
+               return null;
+            }).when(parser).parse(anyObject(), anyObject(), anyObject());
+    NewsContentExtractor newsContentExtractor = new NewsContentExtractor(parser);
+    Optional<NewsArticle> potentialNewsArticle =
+        newsContentExtractor.extractContentFromHtml(
+            webpageStream, NEWS_ARTICLE_WITH_EMPTY_TITLE_CONTENT.getUrl());
+    Assert.assertTrue(potentialNewsArticle.isPresent());
+    Assert.assertEquals(NEWS_ARTICLE_WITH_EMPTY_TITLE_CONTENT, potentialNewsArticle.get());
+  }
+
+  @Test
+  public void extractContentFromHtml_IOExceptionStream() throws IOException {
+    // The webpage stream throws {@code IOException} when being read. This exception should be
+    // caught and no content will be extracted.
+    InputStream badStream = mock(InputStream.class);
+    when(badStream.read(anyObject(), anyInt(), anyInt())).thenThrow(new IOException());
+    Optional<NewsArticle> potentialNewsArticle =
+        realNewsContentExtractor.extractContentFromHtml(badStream, URL);
+    Assert.assertFalse(potentialNewsArticle.isPresent());
+  }
+
+  // SAXException is a checked exception and thus invalid for mocking with {@code thenThrow()}.
+  //
+  // @Test
+  // public void extractContentFromHtml_SAXExceptionStream() throws IOException {
+  //   // The webpage stream throws {@code SAXException} when being read. This exception should be
+  //   // caught and no content will be extracted.
+  //   InputStream badStream = mock(InputStream.class);
+  //   when(badStream.read(anyObject(), anyInt(), anyInt())).thenThrow(new SAXException());
+  //   Optional<NewsArticle> potentialNewsArticle =
+  //       realNewsContentExtractor.extractContentFromHtml(badStream, URL);
+  //   Assert.assertFalse(potentialNewsArticle.isPresent());
+  // }
```

Comment:
Is there a reason there are so many commented-out tests (here and in other files)?